packages/store/src/state-factory.ts

```diff
@@ -101,28 +101,29 @@ export class StateFactory {
   invokeActions(getState, setState, dispatch, actions$, action) {
     const results = [];
 
-    for (const metadata of this.states) {
+    for (const data of this.states) {
+      const metadata: MetaDataModel = data;
+
       const name = getActionTypeFromInstance(action);
-      const actionMetas = metadata.actions[name];
+
+      const actionMetas = metadata.actions[name.toString()];
 
       if (actionMetas) {
         for (const actionMeta of actionMetas) {
           const stateContext = this.createStateContext(getState, setState, dispatch, metadata);
           let result = metadata.instance[actionMeta.fn](stateContext, action);
 
-          if (result === undefined) {
+          if (!result) {
```

Comment:
What if I return false or 0 here? Wouldn’t that break this?

Reply:
so that would be if you just returned a raw value from the state method?